pylutron/__init__.py

```diff
@@ -841,18 +841,127 @@ def handle_update(self, args):
     return False
 
 
-class MotionSensor(object):
+class PowerSource(Enum):
+  """Enum values representing power source, reported by queries to
+  battery-powered devices."""
+  UNSET = -1
+  
+  # Values from ?HELP,?DEVICE,22
+  UNKNOWN = 0
+  BATTERY = 1
+  EXTERNAL = 2
+
+  
+class BatteryStatus(Enum):
+  """Enum values representing battery state, reported by queries to
+  battery-powered devices."""
+  UNSET = -1
+  
+  # Values from ?HELP,?DEVICE,22 don't match the documentation, using what's in the doc.
+  #?HELP says:
+  # <0-NOT BATTERY POWERED, 1-DEVICE_BATTERY_STATUS_UNKNOWN, 2-DEVICE_BATTERY_STATUS_GOOD, 3-DEVICE_BATTERY_STATUS_LOW, 4-DEVICE_STATUS_MIA>5-DEVICE_STATUS_NOT_ACTIVATED>
+  NORMAL = 1
+  LOW = 2
+  OTHER = 3  # not sure what this value means
+
+  
+class MotionSensor(LutronEntity):
   """Placeholder class for the motion sensor device.
 
   Although sensors are represented in the XML, all of the protocol
   happens at the OccupancyGroup level. To read the state of an area,
   use area.occupancy_group.
   """
+
+  _CMD_TYPE = 'DEVICE'
+
+  _ACTION_BATTERY_STATUS = 22
+
+  class Event(LutronEvent):
+    """MotionSensor events that can be generated.
+
+    BATTERY_STATUS_CHANGED: Battery status changed
+        Params:
+          power: PowerSource
+          battery: BatteryStatus
+
+    Note that motion events are reported by OccupancyGroup, not individual
+    MotionSensors.
+    """
+    STATUS_CHANGED = 1
+
   def __init__(self, lutron, name, integration_id):
     """Initializes the motion sensor object."""
-    self._lutron = lutron
-    self._name = name
+    super(MotionSensor, self).__init__(lutron, name)
     self._integration_id = integration_id
+    self._battery = BatteryStatus.UNSET
+    self._power = PowerSource.UNSET
+    self._lutron.register_id(MotionSensor._CMD_TYPE, self)
+    self._query_waiters = _RequestHelper()
+    self._last_update = None
+
+  @property
+  def id(self):
+    """The integration id"""
+    return self._integration_id
+
+  def __str__(self):
+    """Returns a pretty-printed string for this object."""
+    return 'MotionSensor {} Id: {} Battery: {} Power: {}'.format(
+      self.name, self.id, self.battery_status, self.power_source)
+
+  def __repr__(self):
+    """String representation of the MotionSensor object."""
+    return str({'motion_sensor_name': self.name, 'id': self.id,
+                'battery' : self.battery_status,
+                'power' : self.power_source})
+
+  @property
+  def _update_age(self):
+    """Returns the time of the last poll in seconds."""
+    if self._last_update is None:
+      return 1e6
```

Comment:
this confused me at first, should this be float('inf') instead?

Reply:
done